mautic/mautic
# TPROD-339 - SocialBundle - Convert PHP Templates to Twig Templates

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. 



- `/s/monitoring`
  - Does page render with and without data
  - Can be filtered
  - Columns sorted correctly
- `/s/monitoring/new`
  - Form/page render correctly
  - Select each **Monitoring Method** to ensure it updates the form correctly
- `/s/monitoring/view/{id}`
  - Does details page render correctly
- `/s/monitoring/edit/{id}`
  - Form/page renders correctly
- `/s/tweets/new`
  - Form/page renders correctly
- `/s/tweets/edit/{id}`
  - Form/page render correctly
- Within the Camping Builder, add the action **Tweet Contact**, Click the **New Tweet** button. Verify the form renders correctly
  - **NOTE**: There are bugs present on the 5.x branch. Specifically, if you save a tweet it doesn't close the window and add it to the list. Instead it just keeps the window open and redirects to the list page. To see the newly created tweet, you need cancel the new action and add the tweet action again.
- Navigate to to `/s/messages/new`, under the **Tweet** tab, select **Enabled?** and for **Message** select the **Create New** option.
  - Verify the modal renders correctly
- `/s/contacts/view/{id}` **requires app be setup for each social platform**
  - Link the contact to a social account using the integration

**Notes**:
- The social login template is covered by the PR https://github.com/mautic/mautic/pull/11898

**REQUIREMENTS**
- Testing this will require an app be setup for: Facebook, Foursquare, Instagram, LinkedIn, and Twitter. This is only for the contact's profile. The rest of the time, you can use dummy keys.

**Docs**
- https://docs.mautic.org/en/channels/social-monitoring
- https://docs.mautic.org/en/contacts/manage-contacts/contact-overview#social
